tests/components/mqtt_statestream/test_init.py

```diff
@@ -234,12 +245,12 @@ def test_state_changed_event_include_entity(self, mock_utcnow, mock_pub):
 
     @patch("homeassistant.components.mqtt.async_publish")
     @patch("homeassistant.core.dt_util.utcnow")
-    def test_state_changed_event_exclude_domain(self, mock_utcnow, mock_pub):
```

Comment:
We should not alter the behavior of existing tests beyond that is needed to make the function not reject argument changes, it is better to add new ones so we prove the old tests are still working without changes.

Reply:
This one was a bit of laziness on my part. I know all new tests are supposed to be standalone pytests but these were in a class that actually used self so that made it tricky.  Looking at it now though I realize all they're actually  using self for is to get `hass` and there's a fixture for that. So I'll fix it the correct way, sorry about that.

Reply:
Ok so actually its got more reliance on self then I realized. I tried to move it to a standalone pytest but its not really working, its getting into an infinite loop during setup somehow. For now I'm just going to add new tests to the class to get it back to a working state post rebase and feedback rework. If necessary I can try again after.

Reply:
Reverted changes to tests where I added glob filters and added separate ones specifically for that.